Make UseAccessKeyAuthentication public so that it is accessible for p…

FullStackChef opened the pull request:
Make UseAccessKeyAuthentication public so that it is accessible for package extensibility

## Description

When extending on the Azure.Redis package (e.g. https://github.com/CommunityToolkit/Aspire/pull/371) there needs to be a way to determine early (outside of ConfigureInfrastructure) whether the Redis has been configured with AccessKeyAuthentication in order to configure appropriate parameters and outputs

Fixes # (issue)
Have not yet created an issue 
## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [] Yes
  - [x] No
- Did you add public API?
  - [x] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [x] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [x] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [x] No

FullStackChef committed:
Make UseAccessKeyAuthentication public so that it is accessible for package extensibility

FullStackChef committed:
Make UseAccessKeyAuthentication public so that it is accessible for package extensibility

FullStackChef committed:
Make UseAccessKeyAuthentication public so that it is accessible for package extensibility

davidfowl commented:
cc @sebastienros

sebastienros commented:
Since Eric is oof for the next few days, and until we get his opinion, I think we should go with this and we can revert later if there is a better option.

sebastienros commented:
Meaning I am also doing in the [cosmosdb PR](https://github.com/dotnet/aspire/pull/7092)

sebastienros commented:
Maybe the field should be added to `AzurePostgresFlexibleServerResource` for consistency.

FullStackChef commented:
Maybe an IResourceWithMicrosoftEntraAuthentication

FullStackChef commented:
Or an alternative approach could be to introduce a MicrosoftEntraAuthenticatedAnnotation? and have it added by default and removed by WithAccessKeyAuthentication

```csharp
/// <summary>
/// Gets the connection string template for the manifest for the Azure Cache for Redis resource.
/// </summary>
public ReferenceExpression ConnectionStringExpression =>
    this.TryGetLastAnnotation<MicrosoftEntraAuthenticatedAnnotation>(out var _) ?
        ReferenceExpression.Create($"{ConnectionStringOutput}") :
        ReferenceExpression.Create($"{$"{ConnectionStringSecretOutput}"}");
```

sebastienros committed:
Add support for managed identity in Azure Cosmos DB hosting component (#7092)

davidfowl commented:
We can always add an interface later. Focus on the specific type(s) for now

karolz-ms committed:
Restore Helix Docker-related diagnostics info (#7141)

afscrome committed:
Fix typo in `ApplicationExecutor` debug logs (#7142)

davidfowl committed:
Clean up networks, volumes and containers on start of the test. (#7145)

It seems like dcp is still leaking containers in some cases.

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/microsoft/usvc-apiserver build 0.10.2 (#7144)

Microsoft.DeveloperControlPlane.darwin-amd64 , Microsoft.DeveloperControlPlane.darwin-arm64 , Microsoft.DeveloperControlPlane.linux-amd64 , Microsoft.DeveloperControlPlane.linux-arm64 , Microsoft.DeveloperControlPlane.windows-386 , Microsoft.DeveloperControlPlane.windows-amd64 , Microsoft.DeveloperControlPlane.windows-arm64
 From Version 0.10.1 -> To Version 0.10.2

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

paulomorgado committed:
Update Keycloak admin variable names in extensions (#7124)

* Update Keycloak admin variable names in extensions

Renamed constants for Keycloak admin username and password in the `KeycloakResourceBuilderExtensions` class from `KEYCLOAK_ADMIN` and `KEYCLOAK_ADMIN_PASSWORD` to `KC_BOOTSTRAP_ADMIN_USERNAME` and `KC_BOOTSTRAP_ADMIN_PASSWORD`. This change aligns with https://www.keycloak.org/docs/26.0.0/upgrading/#admin-bootstrapping-and-recovery.

* Refactor Keycloak admin environment variables

Removed `KEYCLOAK_ADMIN` and `KEYCLOAK_ADMIN_PASSWORD`
and replaced them with `KC_BOOTSTRAP_ADMIN_USERNAME`
and `KC_BOOTSTRAP_ADMIN_PASSWORD` for improved naming
convention while maintaining functionality.

paulomorgado committed:
Refactor WithRealmImport to accept file or directory (#7121)

* Refactor WithRealmImport to accept file or directory

Updated the `WithRealmImport` method to change the parameter name from `importDirectory` to `import`, allowing it to accept either a directory or a single import file. Adjusted the method logic to handle both cases and updated the XML documentation for clarity.

* Refactor Keycloak tests and update LUTConfig settings

Updated `WithRealmImportShouldThrowWhenBuilderIsNull` test
to check for null import instead of import directory, and
changed the expected exception type for non-existent imports.
Added a new configuration section in `Aspire.lutconfig`
to enable parallel builds, parallel test runs, and set a
test case timeout of 180 seconds.

* Refactor Keycloak import tests and update LUTConfig

Removed obsolete configuration from Aspire.lutconfig.
Added new parameterized tests in KeycloakPublicApiTests.cs
for realm import functionality with directory and file inputs.
Removed the outdated WithRealmImportAddsBindMountAnnotation
test from KeycloakResourceBuilderTests.cs.

paulomorgado committed:
Add health check support for Keycloak container (#7122)

* Add health check support for Keycloak container

This commit introduces new constants for health checks, including
an environment variable (`KC_HEALTH_ENABLED`), a health check
container port (9000), and a health endpoint name (`health`).
The `AddKeycloak` method is updated to incorporate a health
check endpoint and set the health check environment variable
to "true" for the Keycloak resource.

* Refactor Keycloak tests for endpoint annotations and manifest

Updated `AddKeycloakWithDefaultsAddsAnnotationMetadata` to filter `EndpointAnnotation` by name for "http" and "health", adding assertions for the "health" endpoint. Modified `VerifyManifest` to use a specific image reference and added the `KC_HEALTH_ENABLED` environment variable, along with updating the `bindings` section to include the "health" binding.

* Revert change made by mistake.

* Update src/Aspire.Hosting.Keycloak/KeycloakResourceBuilderExtensions.cs

Co-authored-by: Damian Edwards <damian@damianedwards.com>

* Update src/Aspire.Hosting.Keycloak/KeycloakResourceBuilderExtensions.cs

Co-authored-by: Damian Edwards <damian@damianedwards.com>

* Update src/Aspire.Hosting.Keycloak/KeycloakResourceBuilderExtensions.cs

Co-authored-by: Damian Edwards <damian@damianedwards.com>

* Update src/Aspire.Hosting.Keycloak/KeycloakResourceBuilderExtensions.cs

Co-authored-by: Damian Edwards <damian@damianedwards.com>

* Refactor Keycloak resource constants and tests

- Retained `HealthCheckEnvVarName` constant; changed type of `ManagementInterfaceContainerPort` to `int` for clarity.
- Introduced `defaultEndpointName` and `managementEndpointName` variables in tests to replace hardcoded strings for improved readability.
- Updated JSON structure in `VerifyManifest` to change key from "health" to "management" to align with new naming conventions.

---------

Co-authored-by: Damian Edwards <damian@damianedwards.com>
Co-authored-by: David Fowler <davidfowl@gmail.com>

davidfowl committed:
Fix some issues around port forwarding: (#7152)

* Fix some issues around port forwarding:
- Make sure there aren't duplicate labels when written to settings files

- Mark auto forward action for dashboard as open browser and other endpoints as silent.

- Default to hybrid mode for auto port forward. We want to detect ports from the output, but we also want to tear down the portforward when the process dies.

Fixes #7148
Fixes #7147

* Add output so that the dashboard url is always forwarded.

* Single file write
- Attempt to make the end to end more reliable by adding a delay between the file writing and logging the output urls.
- Don't forward any ports that aren't described in settings.

* Clean up
- Rename methods, made AddPortForward sync.
- Added more comments around hacks
- Made a code a little more efficient

* Remove white space

matthebrown committed:
Update service defaults to enable metric exemplars (#7155)

davidfowl committed:
Disable flaky tests (#7179)

- These tests seem to be causing the most trouble on the ci

davidfowl committed:
Don't throw if the docker file or context does not exist (#7165)

- That creation could be deferred and the annotation can be overridden

Fixes #7157

davidfowl committed:
Disable more flaky tests (#7181)

matthebrown committed:
Set EnvironmentCallbackContext.Logger to NullLogger by default (#7180)

davidfowl committed:
Made launch profile annotations public (#7168)

Fixes #7097

vicancy committed:
Adding serviceMode paramter to AddAzureSignalR (#7183)

cqnguy23 committed:
Add support for SignalR emulator (#6793)

* add support for signalr emulator

---------

Co-authored-by: Liangying.Wei <lianwei@microsoft.com>
Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

cqnguy23 committed:
Use withHttpCheck for signalr emulator health check (#7202)

* use native http check for signalr emulator

* remove unnecessary code

davidfowl committed:
Allow the ResourceReadyEvent to block waiters (#7163)

* Allow the ResourceReadyEvent to block waiters
- This allows more easier coordination after a resource is healthy but before waiters are unblocked. This is useful for seeding etc.

davidfowl committed:
Added logs of debug logs to the heath check service (#7209)

JamesNK committed:
Fix dashboard selected resource not updating in details view (#7199)

sebastienros committed:
Move resource creation in ResourceReadyEvent event (#7211)

* Move resource creation in ResourceReadyEvent event

* Update src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

---------

Co-authored-by: David Fowler <davidfowl@gmail.com>

aaronpowell committed:
Deprecating Aspire.Hosting.Dapr (#7198)

* Removing Dapr hosting integration tests

Tests are migrated to Community Toolkit

* Marking all members of the package as obsolete

* Removing the playground dapr project

* Missed an obsolete

eerhardt reviewed (commented):
I think this is OK API to expose. We can expose it on the other Azure resources as well. I don't think we need an interface across those classes at this time.

JamesNK committed:
Add noopener noreferrer nofollow to generated content links (#7117)

JamesNK committed:
Only reset details scroll position if the viewed item changes (#7196)

JamesNK committed:
Default KeyValue with missing to empty string (#7218)

eerhardt committed:
Remove major version updates to runtime, aspnetcore, and efcore packages from dependabot (#7125)

* Remove runtime, aspnetcore, and efcore packages from dependabot

These versions are all split in our dependencies based on TFM, and dependabot doesn't handle that situation well.

* Trying to only update inside major versions

* try without dots

- Add Npgsql
- Add System

* Add update-types under the group

* Add ignore cases as well

karolz-ms committed:
Add detailed Docker container information to pre/post test run commands (#7222)

The detailed info includes network connections in particular, which will help us diagnose lingering "default-aspire-network" networks.

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/microsoft/usvc-apiserver build 0.10.3 (#7224)

[main] Update dependencies from microsoft/usvc-apiserver

FullStackChef committed:
Update src/Aspire.Hosting.Azure.Redis/AzureRedisCacheResource.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

FullStackChef committed:
Update src/Aspire.Hosting.Azure.Redis/AzureRedisCacheResource.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

FullStackChef committed:
Update src/Aspire.Hosting.Azure.Redis/AzureRedisCacheResource.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

FullStackChef commented:
@eerhardt Are there other resources you would like me to update as part of this PR?

eerhardt committed:
Update .NET dependencies to latest versions (#7127)

* Update .NET dependencies to latest versions

- Update runtime, aspnetcore, and efcore dependencies to 9.0.1
- Update dotnet/extensions to 9.1.0

* Updating Version.Details.xml and also update 8.0 era packages

* Update the SDK and runtime previous version for testing

---------

Co-authored-by: Jose Perez Rodriguez <joperezr@microsoft.com>

davidfowl committed:
Clean up processes on start, but wait on shutdown (#7185)

* Clean up processes at the start of the run, not the end
- Added DCP logs to helix output

adamint committed:
Position the language listbox to bottom to fix cutoff (#6972)

* position the language listbox to bottom

* Don't need width property

* Add min-width

---------

Co-authored-by: Adam Ratzman <adamratzman@microsoft.com>

eerhardt committed:
Update Npgsql EF Core to latest 9.0.3 (#7231)

eerhardt commented:
Yes, 2 other places:

https://github.com/dotnet/aspire/blob/fc59d20629343bc9392d3b2f085ae858dbe06fdf/src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresFlexibleServerResource.cs#L34-L35

https://github.com/dotnet/aspire/blob/fc59d20629343bc9392d3b2f085ae858dbe06fdf/src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBResource.cs#L42-L43

dependabot[bot] committed:
Bump the azure group with 11 updates (#7226)

* Bump the azure group with 11 updates

Bumps the azure group with 11 updates:

| Package | From | To |
| --- | --- | --- |
| [Azure.Identity](https://github.com/Azure/azure-sdk-for-net) | `1.13.1` | `1.13.2` |
| [Azure.Messaging.ServiceBus](https://github.com/Azure/azure-sdk-for-net) | `7.18.2` | `7.18.3` |
| [Microsoft.Extensions.Azure](https://github.com/Azure/azure-sdk-for-net) | `1.8.0` | `1.9.0` |
| [Microsoft.Azure.SignalR](https://github.com/azure/azure-signalr) | `1.29.0` | `1.30.0` |
| [Azure.Data.Tables](https://github.com/Azure/azure-sdk-for-net) | `12.9.1` | `12.10.0` |
| [Microsoft.Azure.SignalR.Management](https://github.com/azure/azure-signalr) | `1.29.0` | `1.30.0` |


Updates `Azure.Identity` from 1.13.1 to 1.13.2
- [Release notes](https://github.com/Azure/azure-sdk-for-net/releases)
- [Commits](https://github.com/Azure/azure-sdk-for-net/compare/Azure.Identity_1.13.1...Azure.Identity_1.13.2)

Updates `Azure.Messaging.ServiceBus` from 7.18.2 to 7.18.3
- [Release notes](https://github.com/Azure/azure-sdk-for-net/releases)
- [Commits](https://github.com/Azure/azure-sdk-for-net/compare/Azure.Messaging.ServiceBus_7.18.2...Azure.Messaging.ServiceBus_7.18.3)

Updates `Microsoft.Extensions.Azure` from 1.8.0 to 1.9.0
- [Release notes](https://github.com/Azure/azure-sdk-for-net/releases)
- [Commits](https://github.com/Azure/azure-sdk-for-net/compare/Microsoft.Extensions.Azure_1.8.0...Microsoft.Extensions.Azure_1.9.0)

Updates `Microsoft.Azure.SignalR` from 1.29.0 to 1.30.0
- [Release notes](https://github.com/azure/azure-signalr/releases)
- [Commits](https://github.com/azure/azure-signalr/compare/v1.29.0...v1.30.0)

Updates `Azure.Data.Tables` from 12.9.1 to 12.10.0
- [Release notes](https://github.com/Azure/azure-sdk-for-net/releases)
- [Commits](https://github.com/Azure/azure-sdk-for-net/compare/Azure.Data.Tables_12.9.1...Azure.Data.Tables_12.10.0)

Updates `Microsoft.Azure.SignalR` from 1.29.0 to 1.30.0
- [Release notes](https://github.com/azure/azure-signalr/releases)
- [Commits](https://github.com/azure/azure-signalr/compare/v1.29.0...v1.30.0)

Updates `Microsoft.Azure.SignalR.Management` from 1.29.0 to 1.30.0
- [Release notes](https://github.com/azure/azure-signalr/releases)
- [Commits](https://github.com/azure/azure-signalr/compare/v1.29.0...v1.30.0)

Signed-off-by: dependabot[bot] <support@github.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

JamesNK committed:
Start to split AppExecutor, call events on restart of resources (#7206)

davidfowl committed:
Add snapshot version tracking and improve logging (#7233)

- Added version property to snapshots to avoid missing updates

davidfowl committed:
Build resource logging into Aspire.Hosting (#5878)

* Build resource logging into Aspire.Hosting
- Logging resource state needs to be added before the DcpHostedService is added so that it can subscribe to the resource notification service and capture logs.
- Added EnableResourceLogging to DistributedApplicationOptions
- Removed all of the code duplication and instead set the flag in places where it was being used.

davidfowl committed:
Run this after not before (#7237)

davidfowl committed:
Added support for PublishAsContainerApp to the ExecutableResource (#7234)

Fixes #7223

eerhardt committed:
Update AspNetCore.HealthChecks to 9.0.0 (#7129)

sebastienros committed:
Ignore API version checks in Azurite by default (#7239)

* Ignore API version checks in Azurite by default

* Add missing returns docs

eerhardt committed:
Add role assignment to CosmosDB Entra ID (#7242)

We missed adding it to the infra. Adding the role so it gets put in the bicep file.

JamesNK committed:
Support resource restart in ResourceHealthCheckService (#7238)

* Support resource restart in ResourceHealthCheckService

* Apply suggestions from code review

---------

Co-authored-by: David Fowler <davidfowl@gmail.com>

eerhardt committed:
Change CosmosDB Entra Auth to use data plane roles (#7245)

We currently use the control plane `DocumentDBAccountContributor` role, which doesn't work to add/update/remove items.

Instead of adding the control plane role, assign the data plane contributor role when provisioning the CosmosDB account.

ReubenBond committed:
Add RuntimeUnhealthy to KnownResourceStates (#7212)

mitchdenny committed:
Docs to cover 6006. (#6097)

Co-authored-by: David Fowler <davidfowl@gmail.com>

FullStackChef committed:
Update postgres flexible server resource + unshipped api

FullStackChef committed:
Update postgres flexible server resource + unshipped api

FullStackChef committed:
Update postgres flexible server resource + unshipped api

FullStackChef committed:
Merge branch 'redis-auth-type' of https://github.com/FullStackChef/aspire into redis-auth-type

FullStackChef committed:
Make cosmos db "UseAccessKeyAuthentication" public

FullStackChef committed:
Make cosmos db "UseAccessKeyAuthentication" public

davidfowl commented:
Did you merge with main?

FullStackChef commented:
> Did you merge with main?
Wow @davidfowl  I swear you're like a vampire or something lurking in the shadows!  I hadn't even got the page open yet - I rebased off main. and I've made those additional changes

FullStackChef commented:
@davidfowl and by vampire I meant ninja but covid :(

@eerhardt - I've made the changes you requested.

FullStackChef committed:
Merge branch 'redis-auth-type' of https://github.com/FullStackChef/aspire into redis-auth-type

davidfowl commented:
I think you need to redo this PR, look at the number of changes.

FullStackChef commented:
> I think you need to redo this PR, look at the number of changes.

Thanks will do
